ui/app/components/clients/attribution.js

```diff
@@ -1,45 +1,105 @@
 import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
 
-// TODO: fill out below!!
 /**
  * @module Attribution
- * Attribution components are used to...
+ * Attribution components display the top 10 total client counts for namespaces or auth methods (mounts) during a billing period.
+ * If view is filtered for a single month, two graphs display and show a breakdown of new and total client counts by namespace or auth method, respectively 
  *
  * @example
  * ```js
- * <Attribution @requiredParam={requiredParam} @optionalParam={optionalParam} @param1={{param1}}>
- * Pass in export button
- * </Attribution>
+ * <Clients::Attribution
+ *    @newClientsData={{this.newClientData}}
+ *    @totalClientsData={{this.totalClientData}}
+ *    @chartLegend={{this.chartLegend}}
+ *    @isDateRange={{this.isDateRange}}
+ *    @isAllNamespaces={{this.isAllNamespaces}}
+    >
+      <button type="button"> Export attribution data </button>
+    </Clients::Attribution>
  * ```
- * @param {object} requiredParam - requiredParam is...
- * @param {string} [optionalParam] - optionalParam is...
- * @param {string} [param1=defaultValue] - param1 is...
+
+ * @param {array} newClientsData - (passed to child chart) must be an array of flattened objects
+ * @param {array} totalClientsData - (passed to child chart) must be an array of flattened objects
+ * @param {array} chartLegend - (passed to child) array of objects with key names 'key' and 'label' so data can be stacked
+ * @param {boolean} isDateRange - discerns if dataset from API is a date range or single month
+ * @param {boolean} isAllNamespaces - relays if filtered by all namespaces, or by single namespace
+ * @param {string} activityDateRange - for modal and csv download to display correct date range
  */
 
 export default class Attribution extends Component {
-  get dateRange() {
-    // some conditional that returns "date range" or "month" depending on what the params are
-    return 'date range';
+  @tracked showCSVDownloadModal = false;
+
+  // TODO should this be a getter? It will be updated from parent component's filter bar
```

Comment:
NIT, any chance with TODO you can add your initials in front of them (e.g. CBS TODO or ARG TODO). There are some super old TODOs left over from previous developers, and this way we can show who should be following up with the TODO.

Reply:
Oh my gosh!!! Absolutely!  This whole time I thought your were writing "arg" like "argument" so I left it out because in my vscode makes it nice and orange if I have just have `// TODO` 😆 😆 😆 

These todos will all be cleared up by the time the feature is released, but absolutely will do moving forward!

Reply:
Ha! Someone once thought I had a pirate fetish thing or something. My initials aren't a great example :)